semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TypeOps.scala

```diff
@@ -71,6 +71,7 @@ trait TypeOps {
          */
         case g.WildcardType => s.NoType
         case _: g.BoundedWildcardType => s.NoType
+        case g.ClassInfoType(_, _, sym) if sym.name == g.TypeName("AnyRef") => s.NoType
```

Comment:
I think it's better to ignore than fail in this case, but let me double check what is going on here

Reply:
@tgodzik Also, FWIW, [this](https://github.com/scalameta/scalameta/blob/main/semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/SymbolInformationOps.scala#L108) is where a `TypeBounds` type is being set (since `symbolOf[AnyRef].isAliasType` returns true). When computing the [signature](https://github.com/scalameta/scalameta/blob/main/semanticdb/scalac/library/src/main/scala/scala/meta/internal/semanticdb/scalac/TypeOps.scala#L102) for `TypeBounds` we eventually call `toSemanticTpe` on `slo` which is where the failure occurs.

We could potentially instead try to address this issue upstream of here, e.g., in the `preprocess` method.

Reply:
> chatgpt says the `illegal cyclic reference` occurs when encountering symbol `AnyRef` aliased to `Object` while in a compiler which defines `AnyRef` differently (i.e. deriving from `Any`). basically, clash of different versions of scala. not sure what can be done about this.

@kitbellew Sorry, was this in response to [this](https://github.com/scalameta/scalameta/pull/4085#discussion_r1887098236)?

Reply:
caveat: i don't know anything about this code.

here's what i see: whether we use `TypeBounds(tpe, tpe)` or just `tpe`, `toSemanticTpe` will be called anyway (where you linked, or on line 125 a bit below).

perhaps in `SymbolInformationOps`, in the `preprocess` method, we should call `dealias` (or `dealiasWiden` which looks to be recursive)? i mean, if it's an alias... another option is, in `scala.reflect.internal.Symbols`, there's some code which checks for `isAliasType` and then calls `resultType`, perhaps that's the way.

something like:
```
def preprocess(info: g.Type): g.Type = info.dealiasWiden match {
  case g.PolyType(tparams, tpe) => g.PolyType(tparams, preprocess(tpe))
  case tpe => g.TypeBounds(tpe, tpe)
}
```

Reply:
@kitbellew Ah apologies, I lazily glanced at the git blame and saw your name, but it was just a "formatting commit." It looks like this `preprocess`ing of  `Symbols` which are alias types was added as part of commit `6da0dbd88564` which was introduced in early 2018. Being that this has been in place for so long, and given the committer, I think I'm leaning again towards catching the failure within `toSemanticTpe`, as originally proposed.

Reply:
@tgodzik 

> I think it's better to ignore than fail in this case, but let me double check what is going on here

Have you had a chance to think more about the above?

Reply:
I am looking into that right now, but since the test is not working properly I wend ahead to try and add it to expect tests, but that is somehow broken (most likely by me)

Reply:
Good news is that I actually managed to fix save-expect, bad news this just doesn't compile at all :sweat_smile:

Reply:
Let's just leave the case and give up on the test. It's already fixing things and should not break anything.

It's becoming a bit of waste of time. We can stop using semanticdb on library file altogether in Metals. We wouldn't have that case.

Reply:
@tgodzik Do you want the to generally catch the `ClassInfoType`, or specifically target the case where `ClassInfoType(_, _, <anyref>)`?

Reply:
Just let me know, and I'll clean up the commits and force push so we can get this in :)

Reply:
Generally catch ClassInfoType. We would fail in this case, but I haven't seen that failure being useful ever.

Reply:
I just pushed up the original commit, where we generally catch `ClassInfoType`. Let me know if you disagree

Reply:
Done

Reply:
No worries! It's great that you dig in so deep here, I hope it was at least a bit educational :sweat_smile: We've hit the usual case of trying to compile something we should not have to compile ever since AnyRef is something the compiler creates. This is why it got so complicated